### PR TITLE
update pause image to 3.4.1 and also update the change log

### PR DESCRIPTION
**Pull request by pacoxu**
**What type of PR is this?**
/kind documentation
/sig windows
/kind feature

**What this PR does / why we need it**:
When I build images in my env and it tries to get 3.4.1 which was not built yesterday.
And 

- [x] #91452 pause 3.4.1: Support for Windows container images (OS Versions: 20H2) was added.([#97322](https://prs.k8s.io/97322), [@claudiubelu](https://github.com/claudiubelu))
- [x] #97322 pause 3.4: Support for Windows container images (OS Versions: 1809, 1903, 1909, 2004) was added. ([#91452](https://prs.k8s.io/91452), [@claudiubelu](https://github.com/claudiubelu))
- [x]  #90665 pause 3.3 update debian-base version to v2.1.0 ([#90665](https://prs.k8s.io/90665), [@justaugustus](https://github.com/justaugustus))
- [x] https://github.com/kubernetes/k8s.io/pull/1549 Promote pause:3.4.1 image
- [x] upgrade pause everywhere(in k/k) and make the test pass for e2e and integrations

**Next Step for CRI Runtimes**: CRI downstream changes as only docker-shim's pause image can be configured in kubelet https://github.com/kubernetes/kubernetes/issues/55676#issuecomment-371084007. BTW, dockershim deprecation and plan to deprecate kubelet flag `-pod-infra-container-image`:
- [x] containerd  https://github.com/containerd/containerd/pull/5014
- [x] cri-o update https://github.com/cri-o/cri-o/pull/4550 (containers/common vendor update: https://github.com/containers/common/pull/445)
- [x] kind update  https://github.com/kubernetes-sigs/kind/pull/2062
- [ ] minikube update [search](https://github.com/kubernetes/minikube/search?q=k8s.gcr.io%2Fpause)

**Documentation Change**
- [x] https://github.com/kubernetes/website/issues/26335 and I opened https://github.com/kubernetes/website/pull/28307 for website change

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
```release-note
pause image upgraded to 3.4.1 in kubelet and kubeadm for both Linux and Windows.
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:
```docs

```

### CODE REVIEW

File: cmd/kubeadm/app/constants/constants_windows.go
Comment (neolit123) [inline cmd/kubeadm/app/constants/constants_windows.go:26]: i'm assuming we have to change this too?
https://github.com/kubernetes/kubernetes/blob/d966e0b7fcbe5d0b386f1d99d369841903164599/cmd/kubeadm/app/images/images_windows.go#L33

Comment (neolit123) [reply cmd/kubeadm/app/constants/constants_windows.go:26]: if this is a manifest list image now (multi-arch) we should do the following in kubeadm:

remove these files:
https://github.com/kubernetes/kubernetes/blob/d966e0b7fcbe5d0b386f1d99d369841903164599/cmd/kubeadm/app/images/images_unix.go
https://github.com/kubernetes/kubernetes/blob/d966e0b7fcbe5d0b386f1d99d369841903164599/cmd/kubeadm/app/images/images_windows.go

add this function:
```
// GetPauseImage returns the image for the "pause" container
func GetPauseImage(cfg *kubeadmapi.ClusterConfiguration) string {
	return GetGenericImage(cfg.ImageRepository, "pause", constants.PauseVersion)
}
```

back to:
https://github.com/kubernetes/kubernetes/blob/d966e0b7fcbe5d0b386f1d99d369841903164599/cmd/kubeadm/app/images/images.go

Comment (pacoxu) [reply cmd/kubeadm/app/constants/constants_windows.go:26]: will test it and refactor this if needed

File: build/dependencies.yaml
Comment (ehashman) [inline build/dependencies.yaml:157]: Where is pause 3.4.2 coming from? The rest of this PR is .1? Is this supposed to be the new dev version?

Comment (pacoxu) [reply build/dependencies.yaml:157]: As I don't update the image, I think I should keep it as 3.4.1. 
Will fix it.

**Comment by pacoxu** [comment]
@ehashman @neolit123 Fixed per your comments. 
PTAL again. Thanks.

@claudiubelu Would you take a look at this?

**Comment by claudiubelu** [comment]
So many different pause image versions, this is a nice cleanup.

Greping through this project, these could also be updated:

```
./test/cmd/diff.sh:    kube::test::if_has_string "${output_message}" 'k8s.gcr.io/pause:3.4'
./test/cmd/diff.sh:    kube::test::if_has_string "${output_message}" 'k8s.gcr.io/pause:3.4'
./test/cmd/diff.sh:    kube::test::if_has_string "${output_message}" 'k8s.gcr.io/pause:3.4'
./hack/testdata/pod-changed.yaml:    image: k8s.gcr.io/pause:3.4
```

And maybe this as well:

```
./test/cmd/legacy-script.sh:export IMAGE_PAUSE_V2="k8s.gcr.io/pause:2.0"
```

It seems that the pause image version is also referenced in quite a few other Kubernetes projects, it would take a while to replace all of them.

**Comment by pacoxu** [comment]
@claudiubelu 
the pod-changed.yaml is used to patch the pod and check the image change. So it have to use a different version here.

i keep the legacy script as it use pause_v2 that I am not sure the porposal. so I keep that.

File: cmd/kubeadm/app/images/images_windows.go
Comment (neolit123) [inline cmd/kubeadm/app/images/images_windows.go:33]: we should remove this file and the one for images_unix.go.

basically now that the image location is the same the images_unix.go function GetPauseImage can live under images.go.

**Review by neolit123 — COMMENTED** [review]
/triage accepted
/priority important-longterm

File: cmd/kubeadm/app/images/images.go
Comment (neolit123) [inline cmd/kubeadm/app/images/images.go:133]: the one from _unix.go

```
// GetPauseImage returns the image for the "pause" container
func GetPauseImage(cfg *kubeadmapi.ClusterConfiguration) string {
	return GetGenericImage(cfg.ImageRepository, "pause", constants.PauseVersion)
}
```

Comment (pacoxu) [reply cmd/kubeadm/app/images/images.go:129]: Looks good to me.
It seems to be correct as well.

File: cmd/kubeadm/app/images/images.go
Comment (neolit123) [inline cmd/kubeadm/app/images/images.go:127]:
```suggestion
	return GetGenericImage(cfg.ImageRepository, "pause", constants.PauseVersion)
```

**Review by neolit123 — COMMENTED** [review]
/approve
for kubeadm

File: hack/testdata/filter/pod-apply-selector.yaml
Comment (neolit123) [inline hack/testdata/filter/pod-apply-selector.yaml:11]: i wonder if these are used even. 2.0 was quite some time ago.

i'd recommend leaving all hack/testdata changes out of the PR, but let's see comments from others first.

also these files should be ideally templated instead.

Comment (pacoxu) [reply hack/testdata/filter/pod-apply-selector.yaml:11]: OK. 

I think this will make it easier in the future when we need more integration or e2e tests on windows.

**Review by ehashman — APPROVED** [review]
this seems reasonable to me

**Comment by TBBle** [comment]
Just wanted to note that when this lands, this changes the owner and location of the Windows default pause image from MS to Kubernetes, and [the docs](https://kubernetes.io/docs/setup/production-environment/windows/intro-windows-in-kubernetes/#pause-image) will need to be updated to reflect that.

I noticed this PR because the current MS-supplied pause image does not have a 20H2 build yet, and was trying to work out who owned it to request a new build, but I guess the k8s pause image will make that obsolete soon enough anyway, and I assume I can override `--pod-infra-container-image` in the meantime to point to the multiplatform pause 3.4.1 image in the meantime.

**Comment by pacoxu** [comment]
Thanks, for you reminder. @TBBle, will update docs/website once this is merged.

**Comment by pacoxu** [comment]
Rebase the commits  because PR contains merge commit.

**Review by neolit123 — COMMENTED** [review]
/lgtm
/approve
on my side

File: test/e2e/testing-manifests/scheduling/nvidia-driver-installer.yaml
Comment (cpanato) [inline test/e2e/testing-manifests/scheduling/nvidia-driver-installer.yaml:80]: nit: new line

**Review by cpanato — COMMENTED** [review]
and how about those images defined here:
- https://github.com/kubernetes/kubernetes/blob/master/cluster/gce/windows/smoke-test.sh#L361
- https://github.com/kubernetes/kubernetes/blob/master/cluster/gce/config-common.sh#L161

**Comment by pacoxu** [comment]
Updated. Thanks @cpanato

**Review by cpanato — APPROVED** [review]
/lgtm

**Comment by BenTheElder** [comment]
>  upgrade pause everywhere and make test pass for e2e and integrations

Is it?

Note that we also need to bump it in e.g. containerd, probably cri-o? 
KIND currently prefers the version shipped by the CRI, not the default in dockershim, which I think is actually the correct pattern ...

**Comment by dims** [comment]
@BenTheElder +1 to "KIND prefers the version shipped by the CRI, not the default in dockershim", once we bump all our references we can go evangelize with the CRI impls. then KIND can update.

**Comment by BenTheElder** [comment]
OK, but then we should understand that:
> and make test pass for e2e and integrations

Is not super meaningful, as IIRC we do not run dockershim in presubmit, which is the only thing directly consuming the pause image default setting from in this repo (this value is not passed over CRI, and kubeadm does not configure CRIs).

**Comment by mikebrow** [comment]
Just say when and will make it so in containerd.

**Comment by pacoxu** [comment]
So should we wait for `kind` and `containerd` changes to merge?

- containerd/containerd#5014
- I will check kind change today or tomorrow.

**Comment by BenTheElder** [comment]
> @BenTheElder So should we wait for kind and containerd changes to merge?

I'm not actually sure how we want to approach this.
I think it's new that presubmit doesn't run this from the locations you updatedit, based on dockershim going on the way out.
I don't think we have a great plan for qualifying pause currently, it usually just works fiiiine, but it feels bad to not actually have tests anywhere.

**Comment by dims** [comment]
https://testgrid.k8s.io/sig-node-release-blocking#node-kubelet-master is still testing docker,

```
[dims@dims-a01 22:22] ~/Downloads ⟩ rg 'container-runtime="docker"'
tmp-node-e2e-f61ec3da-ubuntu-gke-1804-1-16-v20200330/kubelet.log
37:Feb 07 01:28:35 tmp-node-e2e-f61ec3da-ubuntu-gke-1804-1-16-v20200330 kubelet[3752]: I0207 01:28:35.342940    3752 flags.go:59] FLAG: --container-runtime="docker"

tmp-node-e2e-f61ec3da-cos-81-12871-1245-6/kubelet.log
38:Feb 07 01:28:33 tmp-node-e2e-f61ec3da-cos-81-12871-1245-6 kubelet[1861]: I0207 01:28:33.020269    1861 flags.go:59] FLAG: --container-runtime="docker"

tmp-node-e2e-f61ec3da-cos-85-13310-1209-7/kubelet.log
38:Feb 07 01:28:33 tmp-node-e2e-f61ec3da-cos-85-13310-1209-7 kubelet[1975]: I0207 01:28:33.349476    1975 flags.go:59] FLAG: --container-runtime="docker"
```

**Comment by pacoxu** [comment]
Are there any projects that I missed? @dims and @BenTheElder 
-  containerd containerd/containerd#5014
-  cri-o update cri-o/cri-o#4550
-  kind update kubernetes-sigs/kind#2062

As https://github.com/kubernetes/kubernetes/issues/98765 mentioned that dockershim is depreacating and we may have a plan to deprecate the `--pod-infra-container-image` in kubelet, we can add some checking in kubelet to check/warning pause image configurations from container info. If cri is using an old pause image, just add a message for warning/infomation.

**Comment by TBBle** [comment]
> Are there any projects that I missed?

https://github.com/kubernetes/minikube has a few references to `k8s.gcr.io/pause:3.2`. Weirdly enough, their [`nvidia-driver-installer` addon ](https://github.com/kubernetes/minikube/blob/6881977b5eadab48801eda86d33cdce2f5475806/pkg/minikube/assets/addons.go#L348-L360) seems to mandate `k8s.gcr.io/pause:2.0`.

**Review by soltysh — APPROVED** [review]
/approve

**Review by spiffxp — APPROVED** [review]
/approve

**Comment by BenTheElder** [comment]
/approve

however: I still think we should write up something about the intended approach to managing this image, so we don't have to determine if it's appropriate on each PR. I'm iffy on the current approach of "upgrade then test" ...

**Pull request merged**
